packages/tools/src/utilities/math/polyline/projectTo2D.ts

```diff
@@ -0,0 +1,48 @@
+import { utilities } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+
+/**
+ * Projects a polyline from 3D to 2D by reducing one dimension.
+ *
+ * @param polyline - The polyline to be projected.
+ * @returns An object containing the shared dimension index and the projected polyline in 2D.
+ * @throws Error if a shared dimension index cannot be found for the polyline.
+ */
+export function projectTo2D(polyline: Types.Point3[]) {
+  // We need to reduce one dimension to 2D, so basically
+  // we need to find the dimension index that is shared by all points
+  // Use the first three points, two is enough but three is more robust
+  let sharedDimensionIndex;
+
+  const testPoints = utilities.getRandomSampleFromArray(
+    polyline,
+    Math.min(50, polyline.length)
+  );
+  for (let i = 0; i < 3; i++) {
+    if (testPoints.every((point, index, array) => point[i] === array[0][i])) {
+      sharedDimensionIndex = i;
+      break;
+    }
+  }
+
+  if (sharedDimensionIndex === undefined) {
+    throw new Error(
+      'Cannot find a shared dimension index for polyline, probably oblique plane'
+    );
+  }
+
+  // convert polyline list and point to 2D
+  const points2D = [] as Types.Point2[];
+
+  for (let i = 0; i < polyline.length; i++) {
+    points2D.push([
+      polyline[i][(sharedDimensionIndex + 1) % 3],
+      polyline[i][(sharedDimensionIndex + 2) % 3],
+    ]);
+  }
```

Comment:
`(sharedDimensionIndex + 1) % 3` and `(sharedDimensionIndex + 2) % 3` never change and can be stored in variables because `%` (mod) is a very expensive operation and that will degrade the performance considerably.

Reply:
If `sharedDimensionIndex` is equal to:
- 0 then [y, z]
- 1 then [z, x] (wouldn't get into an unexpected behavior in this case?)
- 2 then [x, y]

If the answer is yes, I would do:

```javascript
const [firstDim, secondDim] = [0, 1, 2].toSpliced(sharedDimensionIndex, 1)
```